core/src/autogluon/core/searcher/bayesopt/tuning_algorithms/base_classes.py

```diff
@@ -1,10 +1,37 @@
 from abc import ABC, abstractmethod
-from typing import List, Iterator, Iterable, Tuple, Type, Optional, Set, Dict
+from typing import List, Iterator, Iterable, Tuple, Type, Optional, Set, Dict, Union
 import numpy as np
 
 from ..datatypes.common import Candidate
 from ..datatypes.tuning_job_state import TuningJobState
 
+DEFAULT_METRIC = 'active_metric'
+DEFAULT_COST_METRIC = 'cost_metric'
+DEFAULT_CONSTRAINT_METRIC = 'constraint_metric'
+
+
+def dictionarize_objective(x):
```

Comment:
OK, is there not a potential problem here?
- In the code, you allow the user to define the active metric name, and only if not is DEFAULT_METRIC used
- But in dictionarize_objective, DEFAULT_METRIC is hardcoded.

I'd prefer if the hardcoded names are used everywhere, no? There seems no advantage of the user choosing these names, because they are internal anyway.